src/plugins/query_enhancements/public/search/sql_search_interceptor.ts

```diff
@@ -34,16 +33,13 @@ export class SQLSearchInterceptor extends SearchInterceptor {
     signal?: AbortSignal,
     strategy?: string
   ): Observable<IOpenSearchDashboardsSearchResponse> {
-    const { id, ...searchRequest } = request;
     const context: EnhancedFetchContext = {
       http: this.deps.http,
-      path: trimEnd(API.SQL_SEARCH),
+      path: trimEnd(strategy === SEARCH_STRATEGY.SQL_ASYNC ? API.SQL_ASYNC_SEARCH : API.SQL_SEARCH),
       signal,
     };
 
-    const query = this.buildQuery(strategy);
-
-    return fetch(context, query).pipe(
+    return fetch(context, this.queryService.queryString.getQuery()).pipe(
```

Comment:
doesnt the request object already have the whole query? why ae we fetching the query from the service?

Reply:
you are right. this is an existing issue i will need to fix in a subsequent PR.